packages/legend-query-builder/src/stores/milestoning/QueryBuilderMilestoningState.ts

```diff
@@ -86,8 +111,185 @@ export class QueryBuilderMilestoningState implements Hashable {
+              node.propertyExpressionState.derivedPropertyExpressionStates[0]
+                .derivedProperty,
+            )
+          ) {
+            node.propertyExpressionState.derivedPropertyExpressionStates[0]?.parameterValues.forEach(
+              (p) => {
+                if (p instanceof INTERNAL__PropagatedValue) {
+                  p.isPropagatedValue = false;
+                }
+              },
+            );
+          }
+        }
+      });
+    }
+  }
+
   get isMilestonedQuery(): boolean {
-    return Boolean(this.businessDate ?? this.processingDate);
+    return (
+      Boolean(this.businessDate ?? this.processingDate) ||
+      this.queryBuilderState.getAllFunction ===
+        QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS ||
+      this.queryBuilderState.getAllFunction ===
+        QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS_IN_RANGE
+    );
+  }
+
+  get isCurrentClassMilestoned(): boolean {
+    const currentclass = this.queryBuilderState.class;
+    if (currentclass !== undefined) {
+      const stereotype = getMilestoneTemporalStereotype(
+        currentclass,
+        this.queryBuilderState.graphManagerState.graph,
+      );
+      return stereotype !== undefined;
+    }
+    return false;
+  }
+
+  get isCurrentClassSupportsVersionsInRange(): boolean {
+    const currentclass = this.queryBuilderState.class;
+    if (currentclass !== undefined) {
+      const stereotype = getMilestoneTemporalStereotype(
+        currentclass,
+        this.queryBuilderState.graphManagerState.graph,
+      );
+      return (
+        stereotype !== undefined &&
+        stereotype !== MILESTONING_STEREOTYPE.BITEMPORAL
+      );
+    }
+    return false;
+  }
+
+  get isAllVersionsEnabled(): boolean {
+    return (
+      this.queryBuilderState.getAllFunction ===
+        QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS ||
+      this.queryBuilderState.getAllFunction ===
+        QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS_IN_RANGE
+    );
+  }
+
+  get isAllVersionsInRangeEnabled(): boolean {
+    return (
+      this.queryBuilderState.getAllFunction ===
+      QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS_IN_RANGE
+    );
+  }
+
+  get isInavlidAllVersionsInRange(): boolean {
+    if (
+      (this.startDate && !this.endDate) ||
+      (!this.startDate && this.endDate)
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  toggleAllVersionsInRange(val: boolean | undefined): void {
+    if (val) {
+      this.queryBuilderState.setGetAllFunction(
+        QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS_IN_RANGE,
+      );
+      this.initialiseAllVersionsInRangeParameters();
+    } else {
+      this.queryBuilderState.setGetAllFunction(
+        QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS,
+      );
+      this.clearAllVersionsInRangeParameters();
+    }
+  }
+
+  toggleAllVersions(val: boolean | undefined): void {
```

Comment:
these methods arent really toggling, they are more setting the value right? or handling the seting of the turning on/off the feature. can we rename